biorbd_optim/path_conditions.py

```diff
@@ -134,15 +135,26 @@ def regulation(self, nb_elements):
         """
         if len(self.init) == 0:
             self.init = [0] * nb_elements
-        self.regulation_private(self.init, nb_elements, "Init")
+        if self.init_lineaire:
+            self.regulation_private(self.init, 2 * nb_elements, "Init")
+        else:
+            self.regulation_private(self.init, nb_elements, "Init")
 
         if len(self.first_node_init) == 0:
             self.first_node_init = self.init
         if len(self.last_node_init) == 0:
             self.last_node_init = self.init
 
-        self.regulation_private(self.first_node_init, nb_elements, "First node init")
-        self.regulation_private(self.last_node_init, nb_elements, "Last node init")
+        if self.init_lineaire:
+            self.regulation_private(self.first_node_init,
+                                    2 * nb_elements, "First node init")
+            self.regulation_private(self.last_node_init,
+                                    2 * nb_elements, "Last node init")
```

Comment:
This doesn't really make sense since by definition first_node_init is the value for the first node. It cannot hold to values. Same for last node

Reply:
Thank you for your answer.
You are completely right, I will correct it and attribute the value to them.